isis/tests/FunctionalTestsKerneldbgen.cpp

```diff
@@ -0,0 +1,452 @@
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, select2.findKeyword("Time")[1], "2005 DEC 16 00:01:03.241556 TDB");
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, select2.findKeyword("File"), "data/kerneldbgen/mro_sc_2005-12-15.bc");
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, select2.findKeyword("Type"), "Reconstructed");
+}
+
+ /**
+  * Coverage Level Test for kerneldbgen
+  *
+  * This test creates an output database file from the kernel in the input file
+  * that follow the given filter for reconstructed ck file name patterns. A database
+  * is output with time coverage at the SPICE segment level and at the SPICE interval
+  * level. (There will be one output entry for the spice segment, and several for the
+  * SPICE interval because a SPICE segment is composed of SPICE intervals.)
+  *
+  * After the output PVL is created, when compared, the DIFF file indicates to
+  * ignore RunTime and File.  The File keyword is ignored since, depending on
+  * where the test is run, files may have different paths. These paths can not be
+  * removed since they may be long enough to take up multiple lines.
+  *
+  * This test uses files from the TGO CaSSIS mission, as this is where the problem
+  * was identified.
+  *
+  * history 2018-05-09 Kristin Berry - Added test for newly added time coverage
+  *                                 LEVEL=(SEGMENT*, INTERVAL) option. See #5410
+ **/
+TEST(Kerneldbgen, FunctionalTestKerneldbgenCoverageLevelDefault) {
+  QTemporaryDir prefix;
+  QVector<QString> args = {"to="+ prefix.path() + "/kernel_segment.db.pvl",
+                           "type=CK",
+                           "recondir=data/kerneldbgen",
+                           "reconfilter=em16_tgo_sc_??m_*.bc",
+                           "sclk=$tgo/kernels/sclk/em16_tgo_step_????????.tsc",
+                           "lsk=$base/kernels/lsk/naif0012.tls"};
+
+  UserInterface options(APP_XML, args);
+  try {
+    kerneldbgen(options);
+  }
+  catch (IException &e) {
+    FAIL() << "Unable to generate kernel db: " << e.what() << std::endl;
+  }
+  Pvl kerneldbPvl(options.GetFileName("TO"));
+
+  EXPECT_TRUE(kerneldbPvl.hasObject("SpacecraftPointing"));
+
+  PvlGroup select = kerneldbPvl.findGroup("Selection", Pvl::Traverse);
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, select.findKeyword("Time")[0], "2017 MAR 01 23:02:49.287637 TDB");
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, select.findKeyword("Time")[1], "2017 APR 01 00:01:04.267617 TDB");
+
+}
+
+TEST(Kerneldbgen, FunctionalTestKerneldbgenCoverageLevelInterval) {
+  QTemporaryDir prefix;
+  QVector<QString> args = {"to="+ prefix.path() + "/kernel_interval.db.pvl",
+                           "type=CK",
+                           "level=INTERVAL",
+                           "recondir=data/kerneldbgen",
+                           "reconfilter=em16_tgo_sc_??m_*.bc",
+                           "sclk=$tgo/kernels/sclk/em16_tgo_step_????????.tsc",
```

Comment:
sclk files get purged as new ones come in. Can this be modified to have a local sclk?